src/api/app/controllers/webui/package_controller.rb

```diff
@@ -1044,7 +1044,7 @@ def binary_download
     filename = File.basename(params[:filename]) # Ensure it really is just a file name, no '/..', etc.
     repository = Repository.find_by_project_and_name(@project.to_s, params[:repository].to_s)
 
-    url_generator = ::PackageControllerService::URLGenerator.new(project: @project, package: @package,
+    url_generator = ::PackageControllerService::URLGenerator.new(project: @project, package: @package_name,
```

Comment:
I think you forgot to define `@package_name` with `@package_name = params[:package]`

Reply:
Done. Is the `@` in `@package_name` needed?

Reply:
nop, is not needed